### PR TITLE
Adding Publisher and DevPortal comments tests

### PR DESCRIPTION
**Pull request by YasasRangika**
### Purpose
Adding and refactoring the tests for publisher and devportal comments feature.

### Goals
- Refactor existing tests
- Add new tests to validate missing scenarios and flows

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/comments/DevPortalCommentTest.java [hunk 1]
```diff
@@ -169,134 +172,143 @@ In some databases(mysql, oracle), the timestamp format do not store milliseconds
...
+        assertEquals(getRepliesToVerifyPaginationCommenListDTO.getList().get(1).getContent(), "This is a reply 3",
+                "Offset value does not captured");
+    }
 
-        //Edit a comment
-        //Edit the content only
-        HttpResponse editCommentResponse = restAPIStore.editComment(rootCommentIdToAddReplies, apiId,
-                "Edited root comment", "general");
+    @DataProvider (name = "input-data-provider")
+    public Object[][] inputDataProviderMethod(){
```
Comment (hisanhunais) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/comments/DevPortalCommentTest.java:261]: Shall we fix this formatting error?

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/comments/PublisherCommentTest.java [hunk 2]
```diff
@@ -224,80 +231,77 @@ In some databases(mysql, oracle), the timestamp format do not store milliseconds
...
-        assertEquals(getRepliesToVerifyPaginationCommenListDTO.getList().get(0).getContent(), "This is a reply 2", "Offset value does not captured");
-        assertEquals(getRepliesToVerifyPaginationCommenListDTO.getList().get(1).getContent(), "This is a reply 3", "Offset value does not captured");
+        assertEquals(getRepliesToVerifyPaginationCommenListDTO.getList().get(0).getContent(), "This is a reply 2",
+                "Offset value does not captured");
+        assertEquals(getRepliesToVerifyPaginationCommenListDTO.getList().get(1).getContent(), "This is a reply 3",
+                "Offset value does not captured");
+    }
+
+    @DataProvider (name = "input-data-provider")
+    public Object[][] inputDataProviderMethod(){
```
Comment (hisanhunais) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/comments/PublisherCommentTest.java:253]: Shall we fix this formatting error?

**Review by malinthaprasan — APPROVED** [review]

**Review by hisanhunais — APPROVED** [review]

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/comments/DevPortalCommentTest.java (hunk 1 above)
Comment (YasasRangika) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/comments/DevPortalCommentTest.java:261]: I will send this change with https://github.com/wso2/product-apim/pull/11163 PR

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/comments/PublisherCommentTest.java (hunk 2 above)
Comment (YasasRangika) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/comments/PublisherCommentTest.java:253]: I will send this change with https://github.com/wso2/product-apim/pull/11163 PR

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/comments/DevPortalCommentTest.java (hunk 1 above)
Comment (YasasRangika) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/comments/DevPortalCommentTest.java:261]: Fixed with https://github.com/wso2/product-apim/pull/11163/commits/909e61ea2edc8a1e3f88aead2a5213324d94bd84

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/comments/PublisherCommentTest.java (hunk 2 above)
Comment (YasasRangika) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/comments/PublisherCommentTest.java:253]: Fixed with https://github.com/wso2/product-apim/pull/11163/commits/909e61ea2edc8a1e3f88aead2a5213324d94bd84

**Pull request merged**